src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -22,17 +22,30 @@
 #include <ripple/app/main/Application.h>
 #include <ripple/app/misc/Validations.h>
 #include <ripple/app/data/DatabaseCon.h>
+#include <ripple/core/ConfigSections.h>
 #include <boost/format.hpp>
+#include <boost/tokenizer.hpp>
 
 namespace ripple {
 
+namespace detail
+{
+/** Amendments that are "compiled into" this app
+
+    Add amendments to this collection at build time to enable them on this
+   server. The first element of the pair is the id, the second element is
+   the "friendly name".
+*/
+std::set<std::pair<std::string, std::string>> PreEnabledAmendments;
+}
+
 /** Track the list of "amendments"
 
     An "amendment" is an option that can affect transaction processing
```

Comment:
Sorry I missed this before!

First, this seems to be the definition of an "amendment".  But you have comments referring to amendments above it in the file.  This definition needs to appear before its first use.

Second, the sentence should be restructured  - it's the most important piece of documentation in this file so it needs to be clear.

First you should say what an amendment _is_: 
`An "amendment" is an option that can affect transaction processing rules.  Amendments are proposed and then adopted or rejected by the network.`

_Then_ you talk about its services and properties! `An Amendment is uniquely identified by its AmendmentID, a 256-bit key.`

Which leads me to my next point - you shouldn't use a "raw" 256-bit ID for Amendments - you should use a _tagged_ 256-bit type.

An example of how to do it is here:  https://github.com/ripple/rippled/blob/develop/src/ripple/protocol/UintTypes.h#L38

The advantage of doing this is that you are prevented from accidentally assigning a `Directory` back and forth with an `Amendment` even though both of them are 256-bit.

By the way, why did we settle on 256-bit rather than 160-bit?  Not that I particularly care - frankly, I think we should have done 256-bit everything :-D - but I'm curious.